Assets/Scripts/Pathfinding/PathNode.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PathNode
+{
+
+    private List<PathNode> neighbourList;
+
+    public int x;
+    public int y;
+
+    public int gCost;
+    public int hCost;
+    public int fCost;
```

Comment:
Je suis moyen à l'aise d'exposer gCost, hCost et fCost au public, surtout puisqu'elles ne sont utilisées que la par classe Pathfinding. Rien n'empêche un user extérieur de modifier le gCost au beau milieu de nulle part depuis l'autre bout du code, et ça ça va faire chier le pathfinding.
Ce que je ferais, c'est que je garderais PathNode comme un moyen d'interface avec le pathfinding, mais je créerais une structure Cost { qui contient les costs int g, h, f - by the way, les noms sont pas ben ben clair, ça serait pertinent de les renommer en quelque chose de plus explicite } et ensuite un Dictionary<PathNode, Cost> dans la classe Pathfinding elle même, et je jouerais avec ce Dict là seulement. Comme ça, le user n'y a pas accès et peut quand même avoir une référence sur la Node.